ext/opentelemetry-ext-botocore/src/opentelemetry/ext/botocore/__init__.py

```diff
@@ -0,0 +1,208 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+Instrument `Botocore`_ to trace service requests.
+
+There are two options for instrumenting code. The first option is to use the
+``opentelemetry-auto-instrumentation`` executable which will automatically
+instrument your Botocore client. The second is to programmatically enable
+instrumentation via the following code:
+
+.. _Botocore: https://pypi.org/project/botocore/
+
+Usage
+-----
+
+.. code:: python
+
+    from opentelemetry import trace
+    from opentelemetry.ext.botocore import BotocoreInstrumentor
+    from opentelemetry.sdk.trace import TracerProvider
+    import botocore
+
+    trace.set_tracer_provider(TracerProvider())
+
+    # Instrument Botocore
+    BotocoreInstrumentor().instrument(
+        tracer_provider=trace.get_tracer_provider()
+    )
+
+    # This will create a span with Botocore-specific attributes
+    session = botocore.session.get_session()
+    session.set_credentials(
+        access_key="access-key", secret_key="secret-key"
+    )
+    ec2 = self.session.create_client("ec2", region_name="us-west-2")
+    ec2.describe_instances()
+
+API
+---
+"""
+
+import logging
+
+from botocore.client import BaseClient
+from wrapt import ObjectProxy, wrap_function_wrapper
+
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.ext.botocore.version import __version__
+from opentelemetry.trace import SpanKind, get_tracer
+
+logger = logging.getLogger(__name__)
+
+ARGS_NAME = ("action", "params", "path", "verb")
+TRACED_ARGS = ["params", "path", "verb"]
```

Comment:
looks like this variable is only used in the context of trace_args, which only uses it to for an "in" operator. Should this be converted to a set?

Reply:
Fixed